### PR TITLE
Enable hostNetwork support in eck-operator Helm chart

### PR DESCRIPTION
**Pull request by KrisJohnstone**
Enable support for webhooks for those of us whom are using alternate CNIs.

### CODE REVIEW

**Comment by KrisJohnstone** [comment]
> 

I most certainly did sign the agreement.

File: deploy/eck-operator/values.yaml [hunk 1]
```diff
@@ -236,3 +236,6 @@ global:
   # kubeVersion is the effective Kubernetes version we target when generating the all-in-one.yaml.
   kubeVersion: 1.21.0
 
+# When using AWS EKS and an altnerate CNI inconjunction with webhook enabled, hostNetwork needs to enabled to allow for
```
Comment (thbkrkr) [inline deploy/eck-operator/values.yaml:239]:
```suggestion
# HostNetwork allows a Pod to use the Node network namespace.
# This is required to allow for communication with the kube API when using some alternate CNIs in conjunction with webhook enabled.
```

How about moving this just below the `webhook` field L123 where there is currently `softMultiTenancy`?

Comment (KrisJohnstone) [reply deploy/eck-operator/values.yaml]: Works for me, updated the field description and moved the hostNetwork boolean to the webhook section.

File: deploy/eck-operator/values.yaml [hunk 2]
```diff
@@ -119,6 +119,9 @@ webhook:
   # objectSelector corresponds to the objectSelector property of the webhook.
   # Setting this restricts the webhook to act only on objects that match the selector.
   objectSelector: {}
+  # HostNetwork allows a Pod to use the Node network namespace.
+  # This is required to allow for communication with the kube API when using some alternate CNIs in conjunction with webhook enabled.
```
Comment (thbkrkr) [inline deploy/eck-operator/values.yaml:123]:
```suggestion
  # This is required to allow for communication with the kube API when using some alternate CNIs in conjunction with webhook enabled.
  # CAUTION: Proceed at your own risk. This setting has security concerns such as allowing malicious users to access workloads running on the host.
```

File: deploy/eck-operator/templates/statefulset.yaml [hunk 3]
```diff
@@ -117,6 +117,10 @@ spec:
         {{- with .Values.volumes }}
           {{- toYaml . | nindent 8 }}
         {{- end }}
+      {{- if .Values.webhook.hostNetwork }}
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      {{- end }}
```
Comment (naemono) [inline deploy/eck-operator/templates/statefulset.yaml:123]: I initially read `hostNetwork: true` without the preceding `{{- if .Values.webhook.hostNetwork }}`, but I see now what I first was mis-reading. 👍

**Review by naemono — COMMENTED** [review]
I'll be good with this after the suggested CAUTION suggestion from @thbkrkr .

**Comment by thbkrkr** [comment]
buildkite test this

**Comment by thbkrkr** [comment]
@elasticmachine run elasticsearch-ci/docs

**Review by thbkrkr — APPROVED** [review]

**Review by naemono — APPROVED** [review]

**Pull request merged**